### PR TITLE
Add total tally

### LINKED ISSUE #186: Consider focusing on total funds raised
We've talked about deprioritizing individual companies contributions for various reasons (mostly coupled to not wanting to stack rank everyone), but one thing we don't showcase is how much money in total has been raised via the pledge (ala a group celebration). I'm not totally sure where it'd make most sense - as some of it depends on scale, but possibly on both the members and homepage. Could easily imagine an annual tally in the future after year one.

### PR DESCRIPTION
**Pull request by vladh**
Closes #186.

Opening this PR as a place to discuss ideas for implementation details.

Potential issues with this:

* Wordsmithing would be nice.
* We probably want a bit more design work here.
* I think the current phrasing is clear, but maybe it would be worth adding a footnote saying that this is the total of all annual reports of all members we're tracking, including donations made before the Pledge existed.

### CODE REVIEW

**Comment by vladh** [comment]
Iterating on this.




I think we're getting somewhere, but we're going to need a bit of a visual tidy up first, things are getting too busy/messy.

**Comment by vladh** [comment]
@chadwhitacre This is ready for review. I think it's good enough to iterate on later.

File: src/pages/about.astro [hunk 1]
```diff
@@ -216,6 +218,15 @@ const members = filterInactiveMembers(await getMembers());
         >
       </section>
 
+      <section>
+        <h2>How is the grand total amount paid by members to maintainers calculated?</h2>
+
+        <p
+          >The grand total is calculated by adding up all of the payments made by our members across all of their annual
+          reports. Note that some of our members' payments may precede their joining the Pledge.</p
+        >
```
Comment (chadwhitacre) [inline src/pages/about.astro:227]: Can we do latest annual report?

Comment (chadwhitacre) [reply src/pages/about.astro:227]: Latest annual report that is newer than a year old. ideally.

**Comment by vladh** [comment]
@chadwhitacre I've implemented this. I do wonder about something, though. Because this grand total is rolling, it prevents us from displaying how much money we've raised in total. The total is always over some not-clearly-defined period.

Showing a _total_ total amount might be very useful for marketing. For example, we were saying it would be nice to pursue certain milestones, where media coverage might be relevant when we reach eg $5 million.

**Comment by chadwhitacre** [comment]
Hmm ... fair points ... 🤔

I guess my thinking is that, in the same way that we expect companies to pay every year, we should report on a yearly basis. What have you done for me lately? A graph that only ever goes up and to the right is generally low-value information. What if report yearly amounts along these lines?

Q1 - "Last year our members paid maintainers"
Q2-Q4 - "So far this year our members paid maintainers"

**Comment by chadwhitacre** [comment]
We could show the previous period in small text beneath for comparison (fine if out of scope here).

**Comment by vladh** [comment]
Hmm…the vast majority of our members have the end of the year as their dateYearEnding. So “total given this year” might be strongly skewed towards $0, as most members will report towards the end of the year.

**Comment by chadwhitacre** [comment]
As long as the report is filed with us it's fine if the year end is in the future. No?

**Comment by vladh** [comment]
Of course, I'm just thinking that members might not report until later in the year, and the number on the website might be quite low for a good part of the year.

Going to think about this more…

**Comment by chadwhitacre** [comment]
We could also manually compute a number and update it periodically. Make it static instead of dynamic, in other words.

**Comment by vladh** [comment]
That's true, but we would still have to decide how to calculate the number.

Still not sure what the best way forward here is. It seems that some grand all-time total is useful, because it will show us, and help us celebrate, the milestones we've reached — 2 million, 5 million, 10 million…

On the other hand, whether we want to do yearly totals is a valid question.

I've written this tool to visualise our members' reporting periods: [reportviz][reportviz].

Here is a screenshot:



18 out of 25 members report at the end of the year.

[reportviz]: https://github.com/opensourcepledge/osp-tools/tree/main/reportviz

**Comment by vladh** [comment]
Reverted ["TotalTally: only take payments from the latest annual report"](https://github.com/opensourcepledge/opensourcepledge.com/pull/188/commits/5e45211013a96ed7ef7c1daa1b6a7b9b30092ae0), because the numbers it yields are not really possible to interpret.

**Comment by chadwhitacre** [comment]
> It seems that some grand all-time total is useful, because it will show us, and help us celebrate, the milestones we've reached — 2 million, 5 million, 10 million…

Sentry [celebrates ARR milestones](https://blog.sentry.io/building-for-the-fortune-500-000/) (our previous [engagement with Nasdaq](https://www.linkedin.com/posts/getsentry_big-day-thanks-to-our-friends-at-nasdaq-activity-7196592371327787008-O7FP/) ;-). Presumably other companies do as well. I think we'll be more credible if we do the same for Pledge, because it's a more meaningful measurement of impact of the Pledge on the overall vitality of the Open Source ecosystem than total sum of revenue.

Sentry can effectively be classed with Tideways and Private Packagist as having reported on last year, because I reported based on our fiscal year, which ends on January 31. Let's reach out to [Tideways](https://github.com/opensourcepledge/crm/issues/13) and [Private Packagist](https://github.com/opensourcepledge/crm/issues/14). If they are planning on making 2024 payments before EOY as Sentry is, then we effectively have:

- 2023 - 3 reports
- 2024 - 22 reports, with 3 more expected from current members before EOY

I've expanded the CRM to track [renewals](https://github.com/opensourcepledge/crm/issues?q=is%3Aissue+is%3Aopen+label%3Arenewal).

We may need to provide guidance on report timing on the FAQ, and possibly iterate on the JSON format. It seems year ending is maybe less important than renewal date.

**Comment by vladh** [comment]
I rebased this, and updated it to only include members' latest report:



This is just to keep things up to date — we might still want to discuss/change this further.

**Pull request merged**